pkg/client/dtclient/settings_client.go

```diff
@@ -841,3 +862,106 @@ func (d *SettingsClient) Delete(ctx context.Context, objectID string) error {
 
 	return nil
 }
+
+func (d *SettingsClient) GetPermission(ctx context.Context, objectID string) (PermissionObject, error) {
+	path, err := getPermissionPathWithID(objectID)
+	if err != nil {
+		return PermissionObject{}, err
+	}
+
+	resp, err := coreapi.AsResponseOrError(d.client.GET(
+		ctx,
+		path,
+		corerest.RequestOptions{CustomShouldRetryFunc: corerest.RetryIfTooManyRequests},
+	))
+
+	apiError := coreapi.APIError{}
+	// when the API returns a 404 it means that you don't have permission (no-access)
+	if errors.As(err, &apiError) && apiError.StatusCode == http.StatusNotFound {
+		return PermissionObject{}, nil
+	}
+
+	if err != nil {
+		return PermissionObject{}, fmt.Errorf("failed to get permission: %w", err)
+	}
+
+	var result PermissionObject
+	if err = json.Unmarshal(resp.Data, &result); err != nil {
+		return PermissionObject{}, fmt.Errorf("failed to unmarshal permission response: %w", err)
+	}
+
+	return result, nil
+}
+
+func (d *SettingsClient) UpsertPermission(ctx context.Context, objectID string, permission PermissionObject) error {
+	path, err := getPermissionPathWithID(objectID)
+	if err != nil {
+		return err
+	}
+
+	payload, err := json.Marshal(permission)
+	if err != nil {
+		return fmt.Errorf("failed to marshal permission: %w", err)
+	}
+
+	// In the first sep we try to update the remote object
+	_, err = coreapi.AsResponseOrError(d.client.PUT(
```

Comment:
Should we do it 'correctly'? Aka fetch the permissions object, and if it does not exist, then create it, otherwise update it? I know that we have this update-first-and-if-fail-create code multiple times, but imo it's just hacky.